CIPs/CIP-79/cip-79.md

```diff
@@ -0,0 +1,270 @@
+---
+cip: 79
+title: 3ID DID Method Specification
+author: Joel Thorstensson (@oed)
+discussions-to: https://github.com/ceramicnetwork/CIP/issues/80
+status: Draft
+category: Standards
+type: RFC
+created: 2021-02-12
+---
+
+# 3ID DID Method Specification
+
+3ID is a DID method that is implemented natively on Ceramic. It uses a Ceramic document as a source of truth of the information which makes up the DID document for the 3ID. Documents in Ceramic achieves secure key rotation though *proof-of-publication* by anchoring document updates into a blockchain. This means that 3ID inherits this property.
+
+## DID Method Name
+
+The name string that shall identify this DID method is: `3`.
+
+A DID that uses this method MUST begin with the following prefix: `did:3`. Per the [DID specification](https://w3c.github.io/did-core/), this string MUST be in lowercase. The remainder of the DID, after the prefix, is specified below.
+
+## Method Specific Identifier
+
+There are two versions of 3IDs, both versions use a Ceramic document as a way to update the DID document. 3IDv1 is the most recent version, 3IDv0 is supported for legacy reasons. If you are creating new 3IDs you should always use 3IDv1.
+
+### 3IDv1
+
+The method specific identifier for 3IDv1 is simply a Ceramic DocID. The DocID used refers to the Ceramic document which contains the information needed to construct the DID Document upon resolution. DocIDs are encoded according to [CIP-59](https://github.com/ceramicnetwork/CIP/blob/master/CIPs/CIP-59/CIP-59.md).
+
+```
+3idv1 = "did:3:<docId>"
+```
+
+#### Example
+
+```sh
+did:3:kjzl6cwe1jw149tlplc4bgnpn1v4uwk9rg9jkvijx0u0zmfa97t69dnqibqa2as
+```
+
+### 3IDv0
+
+The method specific identifier for 3IDv0 is a [CID](https://github.com/multiformats/cid) as produced by the [IPLD](https://github.com/ipld/specs) codec [dag-cbor](https://github.com/ipld/specs/blob/master/block-layer/codecs/dag-cbor.md).
+
+```
+3idv1 = "did:3:<cid>"
+```
+
+#### Example
+
+```sh
+did:3:bafyreiffkeeq4wq2htejqla2is5ognligi4lvjhwrpqpl2kazjdoecmugi
+```
+
+## CRUD Operation Definitions
+
+In this section the CRUD operations for a 3ID DID are defined.
+
+### Create (Register)
+
+A `3` DID is created by simply creating a Ceramic [`tile`](https://github.com/ceramicnetwork/CIP/blob/master/CIPs/CIP-8/CIP-8.md) document. The [`tile`](https://github.com/ceramicnetwork/CIP/blob/master/CIPs/CIP-8/CIP-8.md) document type takes a DID as the *controller*, and it's recommended that a `did:key` is used. The *controller* is the DID which is allowed to update the document. The *family* of the document is set to `3id`, and the *deterministic* flag is set to `true`.
```

Comment:
I feel like we should say something about what the controller key should be and what it represents and what its relationship is to the keys in the `publicKeys` array in the content body

Reply:
What do you mean with "what the controller key should be"? It doesn't need to have any relationship to the keys in the `publicKeys` array.

Reply:
Right, I'm thinking we should probably say that explicitly, and clarify what the relationship between the controller and the 3id is.  So basically the `publicKeys` are allowed to sign and encrypt messages with this 3id, but cannot change the set of keys that are allowed to do so.  The `controller` is the super-user who can modify the set of keys that can authenticate as the 3id (and therefore could always just grant itself access to authenticate as the 3id if it wanted)

Reply:
> The *controller* is the DID which is allowed to update the document.

This not clear enough?

Added a sentence on the public keys.